src/d3a/models/myco_matcher/myco_external_matcher.py

```diff
@@ -142,10 +147,12 @@ def publish_simulation_id(self, message):
 
     def event_tick(self, **kwargs):
         """Publish the tick event to the Myco client."""
-        is_it_time_for_external_tick = kwargs.pop("is_it_time_for_external_tick", True)
-        # If External matching is enabled, limit the number of ticks dispatched.
-        if not is_it_time_for_external_tick:
+        if (kwargs.get("current_tick_in_slot")
+                and not global_objects.external_global_stats.is_it_time_for_external_tick(
+                kwargs.pop("current_tick_in_slot"))):
+            # If External matching is enabled, limit the number of ticks dispatched.
             return
+        sleep(1)
```

Comment:
What is this `sleep` used for?

Reply:
This is part of the temporary fix/workaround
In the case of the external matching, the `match_recommendations` will be called after allowing myco to send its recommendations
so this slight and adjustable sleep is waiting for the external script to submit recommendations in order to match them in the same tick
Tested using the original bug ticket files

Reply:
@spyrostz It's kind of working, however there are several orders that become invalid at the end of the tick (before clearing in the next tick), even if we do not allow d3a-api-client to replace orders before matching, the update frequency is replacing them anyway